### PR TITLE
LPS-198613 Toggles from system fields from system objects editable after rename field label

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-198613](https://liferay.atlassian.net/browse/LPS-198613)

### CODE REVIEW

**Comment by marcelabc** [comment]
Hey @gabrielwas ,
I'm assigning this PR to you because it is frontend. Feel comfortable assigning to anybody from your squad :+1:

**Comment by marcelabc** [comment]
Hey @cgmonte 
Could you review this PR, please? :smile:

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx [hunk 1]
```diff
@@ -94,7 +94,6 @@ export function RightSidebarObjectFieldDetails() {
 
 			delete objectField.defaultValue;
 			delete objectField.listTypeDefinitionId;
-			delete objectField.system;
```
Comment (cgmonte) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx:97]: Deleting `objectField.system` is ok, we should keep this line as is.

The real problem is in line 88 and 92, above:
```
			if (!editedObjectField) {
				objectField = values;
			}
			else {
				objectField = editedObjectField;
			}
```
Here we are assigning states that are later modified directly, which causes unexpected results. We should assign to a copy before deleting `objectField.system`, like this:.

```
			if (!editedObjectField) {
				objectField = {...values};
			}
			else {
				objectField = {...editedObjectField};
			}
```

This way, we can delete delete `objectField.system` with no unwanted side effects.

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi @Artartur, thanks for working on this! Please check my comment, as I think that the real problem is not having `objectField.system` on `objectField`, but the fact that `objectField` directly references a state object.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx (hunk 1 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx:97]: I see, it makes sense, I've changed what was requested, Thx @cgmonte

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx (hunk 1 above)
Comment (cgmonte) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx:97]: @Artartur

Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx:97]: Yes, I did

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by cgmonte** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**